todo_app/data/mongo_client.py

```diff
@@ -0,0 +1,58 @@
+from flask import session
+from todo_app.data.models import Item, List
+from bson.objectid import ObjectId
+import pymongo
+import os
+
+def get_collection():
+  client = pymongo.MongoClient(os.getenv("MONGODB_CONNECTION_STRING"))
+  db = client[os.getenv("MONGODB_DATABASE_NAME")]
+  collection = db[os.getenv("MONGODB_COLLECTION_NAME")]
+  return collection

```

Comment:
This is a nice solution that neatly avoids having any race conditions with our testing 👍 

You may notice though that the app is _fairly_ slow, and that's actually caused by spinning up a fresh DB client each time. What we can do instead is to implement a simple variation on the singleton pattern that allows us to reuse the first connection, and you'll spot a notable speed increase if we do!

```suggestion

stored_collection = None
def get_collection():
  global stored_collection
  if stored_collection is not None:
    return stored_collection
  client = pymongo.MongoClient(os.getenv("MONGODB_CONNECTION_STRING"))
  db = client[os.getenv("MONGODB_DATABASE_NAME")]
  collection = db[os.getenv("MONGODB_COLLECTION_NAME")]
  stored_collection = collection
  return collection

```

(On most projects, we'd probably _actually_ achieve this by wrapping these functions into a class and having the connection be stored in the constructor, but it's interesting to see an alternative!)